File: src/main/java/org/processmining/models/descriptions/DescriptionWithDataObjects.java
package org.processmining.models.descriptions;

import org.processmining.models.bpmn_with_data.LoggableDataObject;
import org.processmining.models.bpmn_with_data.LoggableStringDataObject;
import org.processmining.models.graphbased.directed.bpmn.BPMNNode;
import org.processmining.models.graphbased.directed.bpmn.elements.Activity;
import org.processmining.models.graphbased.directed.bpmn.elements.Gateway;

import java.util.Collection;
import java.util.Map;
import java.util.Set;

/**
 * Created by Ivan on 22.02.2016.
 */
public interface DescriptionWithDataObjects
{
    Collection<LoggableDataObject> getDataObjects();
    void setDataObjectScriptPaths();
    Set<BPMNNode> getNodesWithOutputDataObjects();
    String getActivityScriptPath(Activity activity);
    Collection<LoggableStringDataObject> getOutputDataObjects(BPMNNode node);
    Set<Gateway> getExclusiveGatewaysWithInputDataObjects();
    Map<Gateway, String> getGatewaysToScriptPaths();
    Collection<LoggableStringDataObject> getInputDataObjects(BPMNNode node);
}

File: src/main/java/org/processmining/models/descriptions/DescriptionWithDataObjects.kt
package org.processmining.models.descriptions

import org.processmining.models.bpmn_with_data.LoggableDataObject
import org.processmining.models.bpmn_with_data.LoggableStringDataObject
import org.processmining.models.graphbased.directed.bpmn.BPMNNode
import org.processmining.models.graphbased.directed.bpmn.elements.Activity
import org.processmining.models.graphbased.directed.bpmn.elements.Gateway

/**
 * Created by Ivan on 22.02.2016.
 */
interface DescriptionWithDataObjects {
    val dataObjects: Collection<LoggableDataObject>
    
    val nodesWithOutputDataObjects: Set<BPMNNode>
    
    val exclusiveGatewaysWithInputDataObjects: Set<Gateway>
    
    val gatewaysToScriptPaths: MutableMap<Gateway, String>
    
    fun setDataObjectScriptPaths()
    
    fun getActivityScriptPath(activity: Activity): String
    
    fun getOutputDataObjects(node: BPMNNode): Collection<LoggableStringDataObject>
    
    fun getInputDataObjects(node: BPMNNode): Collection<LoggableStringDataObject>
}
